src/vsr/replica.zig

```diff
@@ -10212,6 +10271,8 @@ pub fn ReplicaType(
             assert(requests_count <= constants.grid_repair_request_max);
             if (requests_count == 0) return;
 
+            self.repair_messages_budget_grid -|= requests_count;
```

Comment:
Maybe assert that the budget is at least request_max, and than do a plain subtraction? 

I think this'll require our `refil` to also be at least `requests_max`, so that the timeout call-site works-out, but that seems like a good idea?